carbon-design-system/carbon
# fix(tab): tabpanel is tabbable when content has no tabbable content

Closes #7078

~Add `hasNoFocusableContent` prop to Tab to inform if the tabpanel container should be focusable. This preserves existing functionality and allows users to opt-in to the new focus behavior when their tab content is text only or has no focusable elements/content.~

Add `useTabbableContent` to TabContent to automatically determine if the contents contains a tabbable element.

#### Changelog

**New**

- Add `useTabbableContent` hook to TabContent

**Changed**

- Update Tabs story to have a tabbable button in the contents of one tab
- Refactor TabContent tests to use testing-library and test for tabindex/hook functionality

#### Testing / Reviewing

1. Open the [default Tabs story](https://deploy-preview-7894--carbon-components-react.netlify.app/?path=/story/components-tabs--default)
1. Type the <kbd>Tab</kbd> key - focus should move to the tabpanel/content container.
1. Type <kbd>Shift + Tab</kbd> - focus moves back to "Tab Label 1"
1. Type <kbd>→</kbd> - focus should move to "Tab Label 2"
1. Type the <kbd>Tab</kbd> key again - focus should move to the button inside the content (the tabpanel should not be focused as the content contains a focusable element)
